jetson/nav/search/searchStateMachine.cpp

```diff
@@ -96,11 +100,82 @@ NavState SearchStateMachine::executeSearchSpin()
             return NavState::SearchTurn;
         }
         nextStop += waitStepSize;
-        return NavState::SearchSpinWait;
+        return NavState::SearchWait;
     }
     return NavState::SearchSpin;
 } // executeSearchSpin()
 
+//Executes the logic for a gimbal search spin. The main objective of a gimbal search spin is to spin the gimbal
+//to positive "gimbalSearchAngleMag" (150) then to -150 then to 0. Every "wait step size" we stop the gimbal
+//in order to give it time to find the target. Only use the wait step size on the 1st phase of the rotation
+//this is because its unnecsesary to double check
+NavState SearchStateMachine::executeSearchGimbal( )
+{
```

Comment:
this code looks almost exactly like the code from gateStateMachine.cpp...like is there any way to make these large reused parts into a general object or something both these classes inherit from? im concerned in debugging things in gate those fixes will be missed here/visa versa